clients/js/src/das.ts

```diff
@@ -0,0 +1,318 @@
+          creators: creators.map((creator: any) => ({
+            ...creator,
+            address: publicKey(creator.address),
+          })),
+        }
+      : {}),
+    ...(rule_set !== undefined ? { ruleSet: getRuleSet(rule_set) } : {}),
+    ...(attribute_list !== undefined ? { attributeList: attribute_list } : {}),
+    ...(frozen !== undefined ? { frozen } : {}),
+  }))(dasPluginData);
+}
+
+export function dasPluginsToCorePlugins(dasPlugins: Record<string, any>) {
+  return Object.keys(dasPlugins).reduce((acc: PluginsList, dasPluginKey) => {
+    const dasPlugin = dasPlugins[dasPluginKey];
+    const { authority, data, offset } = dasPlugin;
+    const authorityAddress = authority?.address;
+
+    acc = {
+      ...acc,
+      [convertSnakeCase(dasPluginKey)]: {
+        authority: {
+          type: authority.type,
+          ...(authorityAddress ? { address: publicKey(authorityAddress) } : {}),
+        },
+        ...dasPluginDataToCorePluginData(data),
+        offset: BigInt(offset),
+      },
+    };
+
+    return acc;
+  }, {});
+}
+
+export function dasAssetToCoreAssetOrCollection(
+  dasAsset: DasApiAsset
+): AssetV1 | CollectionV1 {
+  // TODO: Define types in Umi DAS client.
+  const {
+    interface: assetInterface,
+    id,
+    ownership: { owner },
+    content: {
+      metadata: { name },
+      json_uri: uri,
+    },
+    compression: { seq },
+    grouping,
+    authorities,
+    plugins,
+    executable,
+    lamports: lamps,
+    rent_epoch: rentEpoch,
+    mpl_core_info: mplCoreInfo,
+  } = dasAsset as DasApiAsset & {
+    plugins: Record<string, any>;
+    executable?: boolean;
+    lamports?: number;
+    rent_epoch?: number;
+    mpl_core_info?: {
+      num_minted: number;
+      current_supply: number;
+      plugins_json_version: number;
+    };
+  };
+  const { num_minted: numMinted = 0, current_supply: currentSize = 0 } =
+    mplCoreInfo ?? {};
+  const commonFields = {
+    publicKey: id,
+    uri,
+    name,
+    ...getAccountHeader(executable, lamps, rentEpoch),
+    ...dasPluginsToCorePlugins(plugins),
+    // pluginHeader: // TODO: Reconstruct
+  };
+
+  const isCollection =
+    assetInterface === (MPL_CORE_COLLECTION as DasApiAssetInterface);
+  if (isCollection) {
+    return {
+      ...commonFields,
+      key: Key.CollectionV1,
+      updateAuthority: authorities[0].address,
+      numMinted,
+      currentSize,
+    };
+  }
+
+  return {
+    ...commonFields,
+    key: Key.AssetV1,
+    owner,
+    seq: seq ? some(BigInt(seq)) : none(),
+    ...getUpdateAuthority(grouping[0], authorities[0]),
+  };
+}
+
+export async function searchAssets(
+  context: Umi,
+  input: Omit<SearchAssetsRpcInput, 'interface' | 'burnt'> & {
+    interface: typeof MPL_CORE_ASSET | typeof MPL_CORE_COLLECTION;
+  }
+) {
+  const dasAssets = await context.rpc.searchAssets({
+    ...input,
+    interface: input.interface as DasApiAssetInterface,
+    burnt: false,
+  });
+
+  return dasAssets.items.map((dasAsset) =>
```

Comment:
should we filter out items that aren't the type we are looking for? this might be needed for das providers that have not implemented Core yet and may return all asset types

Reply:
Not sure, TBH. As I understood, the idea of this whole DAS integration was to make MPL Core specific stuff only. So, based on this understanding, I'm working with mpl assets only here. Let me know if we need some changes/additions here.

Reply:
Also, if we return other assets there, what should we do with them, return them as is? If so, I'm a bit confused. Then, we'll have data structure inconsistency. MPL Core assets will be transformed, but the rest won't. And if somebody needs assets other than MPL Core ones, why would they use MPL Core DAS functions in the first place?